Plugins~/Src/MeshSyncClientBlender/msblenContext.cpp

```diff
@@ -1326,7 +1326,22 @@ void msblenContext::logInfo(const char * format, ...)
 bool msblenContext::isServerAvailable()
 {
     m_sender.client_settings = m_settings.client_settings;
-    return m_sender.isServerAvaileble();
+
+    int server_session_id;
+    bool available = m_sender.isServerAvailable(&server_session_id);
+
+    // If server session changed, reset texture manager so textures are sent again:
+    if (m_sender.server_session_id != ms::InvalidID &&
+        server_session_id != m_sender.server_session_id) {
+        if (server_session_id != ms::InvalidID) {
+            resetMaterials();
+            // Ensure a full sync:
+            m_server_requested_sync = true;
+        }
+    }
+    m_sender.server_session_id = server_session_id;
```

Comment:
This is a polling function, is it possible to keep this as it is and move this code somewhere else? can we not use this in the sendObjects callback?